app/blueprints/self_serve/data/data_access.py

```diff
@@ -65,21 +66,124 @@ def save_page(page: dict):
 def get_list_by_id(id: str) -> dict:
     return LISTS.get(id, None)
 
-
-def save_question(question: dict):
+# from app.db.queries.application import insert_new_section
+# from app.db.queries.application import insert_new_form
+# from app.db.queries.application import insert_new_page
+# from app.db.queries.application import insert_new_component
+
+def save_template_component(component: dict):
+    """
+    Save a template component to the database
+        Parameters:
+            component: dict    The component to save to the database as a template
+        Returns:
+            dict           The saved component
+    """
+    
+    """ temp in memory solution """
     COMPONENTS.append(
         {
             "json_snippet": {
                 "options": {},
-                "type": question["question_type"],
-                "title": question["title"],
-                "hint": question["hint"],
+                "type": component["question_type"],
+                "title": component["title"],
+                "hint": component["hint"],
             },
-            "id": question["id"],
-            "builder_display_name": question["builder_display_name"],
+            "id": component["id"],
+            "builder_display_name": component["builder_display_name"],
         }
     )
+    """ end temp """
+
+    # component_config = {
```

Comment:
Can we remove the comments ? is it auto generated ?

Reply:
i should of mentioned that this PR was the backend part only, not the routing or frontend as we dont have the designs for this yet, the comments are just placeholders really so ill add the TODOs 👍

Reply:
I think adding todo is enough for now